### ISSUE TITLE
NewType regression in 1.1.402

### ISSUE THREAD

**Issue by bellini666** [microsoft/pyright#10608]
Take the following code as an example:

```python
from typing import NewType, reveal_type

MyStr = NewType("MyStr", str)


reveal_type(MyStr)
reveal_type(MyStr("hello"))


def func(x: type): ...


func(MyStr)
```

Until 1.1.401 this code would produce:

```
/private/tmp/pyright_test/xxx.py
  /private/tmp/pyright_test/xxx.py:6:13 - information: Type of "MyStr" is "type[MyStr]"
  /private/tmp/pyright_test/xxx.py:7:13 - information: Type of "MyStr("hello")" is "MyStr"
0 errors, 0 warnings, 2 informations
```

In 1.1.402, when trying to pass that type to a function expecting a `type`, it says that `"FunctionType" is not assignable to "type"`

```
/private/tmp/pyright_test/xxx.py
  /private/tmp/pyright_test/xxx.py:6:13 - information: Type of "MyStr" is "type[MyStr]"
  /private/tmp/pyright_test/xxx.py:7:13 - information: Type of "MyStr("hello")" is "MyStr"
  /private/tmp/pyright_test/xxx.py:13:6 - error: Argument of type "type[MyStr]" cannot be assigned to parameter "x" of type "type" in function "func"
    "FunctionType" is not assignable to "type" (reportArgumentType)
1 error, 0 warnings, 2 informations
```

**Comment by erictraut** [comment]
This change was intentional. It was required to conform with the typing spec. See https://github.com/microsoft/pyright/discussions/10596 for details.

The runtime value returned by `NewType` is not actually an instance of `type`. It's a function object. It can be called, but it can't be used in other locations where a type can be used, such as in an `isinstance` call.
